### ISSUE TITLE
Usability

### ISSUE THREAD

**Issue by FlorianHeigl** [Normation/rudder-api-client#11]
Hi,

been working with this some more and also found the original release mail, where you asked for feedback. So here's some more feedback!
The tool is helpful but still too complex to get adopted. But it should be!

## Install
- i already opened a redmine issues since i think this should be included in every rudder master install
- likewise - why does one need to manually set it up? if I run setup.py it could install /usr/local/bin/rudder-cli and create a stub cli dotfile.

## Parsing
- Input from jq should be easily used. Currently the parser doesn't recognize the UUIDs from jq queries. The problem is quotes! It considers "123-456-789" something else than 123-456-789.

## Docs:
- Examples need to be more in numbers
- Examples need to be more in complexity
- Examples need to be more in completeness

Currently the documentation is about 5% of how much there should be.

### As a starter:
Showing how to list directives or how to add one by ID is merely syntax help, but not enough as documentation/example. Instead, try an example that accepts all pending nodes matching a certain patterns. (So it needs to feed back the output from the first call into another one)

The directive and group handling doc is also too little. So you're told you need to give those 3 fields and one that is named "parameters". Put an example there that lists directives, shows one of them by ID, deletes it, and adds it back.
Then the same with an update.

## Usage
The final UX pain is the use of IDs. We all know Rudder uses UUIDs internally, but the CLI should really accept either IDs or hostnames, translating after verifying there is no conflict.

Think of a scripted interface between multiple CLI tools. Every single script interfacing with would need code to do the node+uuid listing, and the mapping of both, and the error handling for conflicts. Conflicts that will in 99% arise because of stale inventories *in rudder*.
As a user I'd totally expect to be able to say "rudder-cli node remove myhost.tld" instead of what I'm actually forced to do (write a script that does all the fetching / mapping or do shitty greps with jq, remove the quotes etc etc.)

## tl;dr of feedback:
At the moment it's like a tech demo, but it deserves to be a core feature.
A days of polishing could make it work.

**Comment by matya** [comment]
I would add that if it would be perl ;)

**Comment by peckpeck** [comment]
Thanks for the comment.
The CLI should indeed be a lot better and we should work on it.
However we have a lot of thing with a higher priority for 4.0
We will work on this more in depth for 4.1
